File: Loenn/triggers/gokuro_spawn_trigger.lua
local gokuroSpawn = {}

gokuroSpawn.name = "Sardine7/GokuroSpawnTrigger"

gokuroSpawn.fieldInformation = {
    yApproachSpeed = {
        fieldType = "number",
        minimumValue = 0.0
    },
    attackMaxSpeed = {
        fieldType = "number",
        minimumValue = 0.0
    },
    chronosBarrier = {
        fieldType = "number",
        minimumValue = 0.0
    },
    yPosition = {
        fieldType = "number",
        minimumValue = -1.0
    },
    volume = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 1.0
    },
    attackCount = {
        fieldType = "integer",
        minimumValue = -1
    }
}

gokuroSpawn.placements = {
    name = "gokuro_spawn",
    data = {
        chaseWaitTimes = "1.0,2.0,3.0,2.0,3.0",
        bouncy = true,
        yApproachDuringRespawn = false,
        yApproachSpeed = 100.0,
        dieFast = false,
        attackMaxSpeed = 500.0,
        chronosBarrier = 200.0,
        luigi = false,
        extraApproach = true,
        yPosition = -1.0,
        volume = 1.0,
        attackCount = -1
    }
}

return gokuroSpawn

File: Ahorn/triggers/GokuroSpawnTrigger.jl
module Sardine7GokuroSpawnTrigger

using ..Ahorn, Maple

@mapdef Trigger "Sardine7/GokuroSpawnTrigger" GokuroSpawnTrigger(x::Integer, y::Integer, width::Integer=16, height::Integer=16, chaseWaitTimes::String="1.0,2.0,3.0,2.0,3.0", bouncy::Bool=true, yApproachDuringRespawn::Bool=false, yApproachSpeed::Real=100.0, dieFast::Bool=false, attackMaxSpeed::Real=500.0, chronosBarrier::Real=200.0, luigi::Bool=false, extraApproach::Bool=true, yPosition::Real=-1.0, volume::Real=1.0, attackCount::Integer=-1)

const placements = Ahorn.PlacementDict(
    "Gokuro (Spawn, Sardine7)" => Ahorn.EntityPlacement(
        GokuroSpawnTrigger,
        "rectangle"
    )
)

end